.github/workflows/automated_tests_appium_ios.yml

```
@@ -0,0 +1,127 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Beagle Appium iOS Tests
+
+#on: workflow_dispatch
+# testing for this PULL REQUEST only. Will be set back to 'workflow_dispatch' afterwards 
+on:
+    push:
+        branches:
+            - master
+        paths:
+            - '.github/**'
+    pull_request:
+        paths:
+            - '.github/**'
+
+jobs:
+     
+    appium_tests:
+        
+        name: Run Appium iOS tests
+        runs-on: macos-latest 
+        steps:
+
+            -   uses: actions/checkout@v2
+
+            ##################################################################################################
+            # Caching - Start
+            ##################################################################################################
+
+            -   name: Cache fastlane dependences
+                uses: actions/cache@v2
+                env:
+                    fastlane-cache-key: fastlane-cache-test
+                    fastlane-path: ~/.gem
+                with:
+                     path: ${{ env.fastlane-path }}
+                     key: ${{ runner.os }}-build-${{ env.fastlane-cache-key }}-${{ hashFiles('Gemfile.lock') }}
+
+            -   name: Cache Beagle backend gradle dependencies
+                uses: actions/cache@v2
+                env:
+                     gradle-cache-key: backend-tests-gradle-cache
+                     gradle-path: ~/.gradle
+                with:
+                     path: ${{ env.gradle-path }}
+                     key: ${{ runner.os }}-build-${{ env.gradle-cache-key }}-${{ hashFiles('backend/buildSrc/**') }}
+                     restore-keys: ${{ runner.os }}-build-${{ env.gradle-cache-key }}
+
+            ##################################################################################################
+            # Caching - end
+            ##################################################################################################
+
+            -   name: Install Fastlane
+                run: bundle config set path '~/.gem' && bundle install
+
+            -   name: Start backend in background
+                run: bash fastlane/automatedTests/start_test_backend.sh && sleep 30
+
+            -   name: Config and run Appium server
+                run: bash fastlane/automatedTests/appium/config_and_run_appium_server.sh
+
+            -   name: Build app-ios project to generate the app file
+                run: | 
+                     set -e
+                     cd tests/appium/app-ios/AppiumApp
+                     xcodebuild -showsdks
+                     xcodebuild -sdk iphonesimulator14.2   
+                     cd $GITHUB_WORKSPACE
```

this change will make it possible to access all dependencies that `AppiumApp` target needs through the Xcode workspace.
```suggestion
                run: | 
                     set -e
                     xcodebuild -showsdks
                     xcodebuild  -workspace iOS/Beagle.xcworkspace -scheme AppiumApp -destination 'platform=iOS Simulator,name=iPhone 11' -quiet
```